src/main/java/io/smartcat/cassandra/diagnostics/QueryProcessorWrapper.java

```diff
@@ -123,11 +123,14 @@ private ResultMessage originalProcessStatement(CQLStatement statement, QueryStat
    */
   private void report(final long startTime, final long execTime, final CQLStatement statement,
       final QueryState queryState, final QueryOptions options, final String errorMessage) {
+      QueryReport report = new QueryReport(startTime,
+            execTime,
+            queryState.getClientState().getRemoteAddress().toString(),
+            statement.getClass().getSimpleName());
+
+    logger.trace("Reporting slow query: {}.", report);
     executor.submit(() -> {
-      reporter.report(new QueryReport(startTime,
```

Comment:
The idea in QueryProcessorWrapper::report is to have as little activities as possible and to postpone everything to the executor service, that it, to the a new thread of execution (of lesser priority), in order to lower the performance impact to the original code. That was the reason why the construction of QueryReport is kept inside of executor.submit().